Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1609,18 +1609,18 @@ RefPtr<API::Object> WebProcessProxy::transformHandlesToObjects(API::Object* obje
 
             case API::Object::Type::PageHandle:
                 ASSERT(static_cast<API::PageHandle&>(object).isAutoconverting());
-                return m_webProcessProxy.webPage(static_cast<API::PageHandle&>(object).pageProxyID());
+                return CheckedRef { m_webProcessProxy }->webPage(static_cast<API::PageHandle&>(object).pageProxyID());
```

Comment:
I don't really understand why we're constructing a CheckedRef here. Isn't m_webProcessProxy already a CheckedRef?

Reply:
Because m_webProcessProxy can get cleared during a non-trivial function call, or rather it's not obvious that it would never be. See https://lists.webkit.org/pipermail/webkit-dev/2023-August/032554.html

Reply:
I'll update this based on the latest discussion in the thread.